contracts/pow/pow.cpp

```diff
@@ -23,70 +29,185 @@ struct pow_signature_data
 
 KOINOS_REFLECT( pow_signature_data, (nonce)(recoverable_signature) )
 
-struct difficulty_metadata
+struct difficulty_metadata_v1
 {
    uint256_t      difficulty_target = 0;
    timestamp_type last_block_time = timestamp_type( 0 );
    timestamp_type block_window_time = timestamp_type( 0 );
    uint32_t       averaging_window = 0;
 };
 
-KOINOS_REFLECT( difficulty_metadata,
+KOINOS_REFLECT( difficulty_metadata_v1,
    (difficulty_target)
    (last_block_time)
    (block_window_time)
    (averaging_window)
 )
 
-difficulty_metadata get_difficulty_meta()
+struct difficulty_metadata_v2
 {
-   difficulty_metadata diff_meta;
-   system::db_get_object( contract_id, DIFFICULTY_METADATA_KEY, diff_meta );
-   if ( diff_meta.difficulty_target == 0 )
+   uint256_t      difficulty_target = 0;
+   timestamp_type last_block_time = timestamp_type( 0 );
+   uint64_t       required_hc = 0;
+   uint128_t      hc_reserve;
+   uint128_t      us_reserve;
+   timestamp_type target_block_interval = timestamp_type( TARGET_BLOCK_INTERVAL_MS / 1000 );
+};
+
+KOINOS_REFLECT( difficulty_metadata_v2,
+   (difficulty_target)
+   (last_block_time)
+   (required_hc)
+   (hc_reserve)
+   (us_reserve)
+   (target_block_interval)
+)
+
+using difficulty_metadata = difficulty_metadata_v2;
+
```

Comment:
This `using difficulty_metadata` should be eliminated, and replace `difficulty_metadata` with `difficulty_metadata_v2` in the following code.  The `using` makes the code a little harder to follow than simply being explicit everywhere.